internal/service/device.go

```diff
@@ -423,7 +423,9 @@ func (h *ServiceHandler) DecommissionDevice(ctx context.Context, request server.
 	if deviceObj.Spec != nil && deviceObj.Spec.Decommissioning != nil {
 		return nil, fmt.Errorf("device already has decommissioning requested")
 	}
+
 	deviceObj.Spec.Decommissioning = request.Body
+	deviceObj.Metadata.Owner = nil
```

Comment:
can you elaborate on why this change?

Reply:
IIUC, this `Metadata.Owner` field is what sets the device as part of a fleet. A decommissioning device should no longer be shown as part of a fleet as it is no longer "live" or able to do anything or be edited in any way. At least, that is how we had discussed it.

Reply:
lets add some docs here for our future selves :)